File: atcoder/abc391a_lucky_direction.cpp
// Vicfred
// https://atcoder.jp/contests/abc391/tasks/abc391_a
// implementation
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main() {
  string D;
  cin >> D;
  map<string,string> opposite;
  opposite["N"] = "S";
  opposite["S"] = "N";
  opposite["W"] = "E";
  opposite["E"] = "W";
  opposite["NE"] = "SW";
  opposite["NW"] = "SE";
  opposite["SE"] = "NW";
  opposite["SW"] = "NE";
  cout << opposite[D] << endl;
  return 0;
}

File: atcoder/abc391a_lucky_direction.hs
-- vicfred
-- https://atcoder.jp/contests/abc391/tasks/abc391_a
-- implementation
main :: IO ()
main = getLine >>= putStrLn . f

f :: String -> String
f "N" = "S"
f "E" = "W"
f "W" = "E"
f "S" = "N"
f "NE" = "SW"
f "NW" = "SE"
f "SE" = "NW"
f "SW" = "NE"
